tests/performance_tests/log_reader.py

```diff
@@ -70,3 +78,51 @@ def scrapeLog(total, path):
                         total.updateTotal(int(value[2]), 0, 0, 0, 0, int(v2Logging[0]))
                 else:
                     print("Error: Unknown log format")
+
+def findPrunedRangeOfInterest(data: chainData, numAddlBlocksToDrop=0):
+
+    startBlockIndex = 0
+    endBlockIndex = len(data.blockLog) - 1
+
+    #skip leading empty blocks in initial range of interest as well as specified number of potentially non-empty blocks
+    droppedBlocks = 0
+    for block in data.blockLog:
+        if block.blockNum < data.startBlock or (droppedBlocks == 0 and block.transactions == 0):
+            continue
+        else:
+            if droppedBlocks < numAddlBlocksToDrop:
+                droppedBlocks += 1
+                continue
+            else:
+                startBlockIndex = data.blockLog.index(block)
+                break
+
+    #skip trailing empty blocks at end of initial range of interest as well as specified number of potentially non-empty blocks
+    droppedBlocks = 0
+    for block in reversed(data.blockLog):
+        if block.blockNum > data.ceaseBlock or (droppedBlocks == 0 and block.transactions == 0):
+            continue
+        else:
+            if droppedBlocks < numAddlBlocksToDrop:
+                droppedBlocks += 1
+                continue
+            else:
+                endBlockIndex = data.blockLog.index(block)
+                break
+
+    return startBlockIndex,endBlockIndex
+
+def scoreTransfersPerSecond(data: chainData, numAddlBlocksToDrop=0) -> stats:
+    consecutiveBlockTps = []
+
+    startBlockIndex,endBlockIndex = findPrunedRangeOfInterest(data, numAddlBlocksToDrop)
+
+    if startBlockIndex >= endBlockIndex:
+        print(f"Error: Invalid block index range start: {startBlockIndex} end: {endBlockIndex}")
+        return stats()
+
+    for i in range(startBlockIndex, endBlockIndex+1):
+        if i + 1 < endBlockIndex:
+            consecutiveBlockTps.append(data.blockLog[i].transactions + data.blockLog[i+1].transactions)
+
+    return stats(numpy.min(consecutiveBlockTps) , numpy.max(consecutiveBlockTps), numpy.average(consecutiveBlockTps), numpy.std(consecutiveBlockTps))
```

Comment:
I suggest including in the statistics the number of blocks considered in the statistics and the number of blocks with zero transactions as a check.  My suggested implementation will omit blocks with zero transactions which occur in the middle of the test run.  As implemented today, none should exist, but that could change.  Having the total block count and the empty block count should be generally useful for seeing if a run behaved as expected.

Reply:
The additional data points in the `stats` is a great idea.  I'll add that.

`As implemented today, none should exist` -- I disagree here, and that is why it is implemented the way it is. Blocks with zero transactions in the middle of a test run should not be filtered out as that is a very interesting data point to consider.